lab_3_generate_by_ngrams/main.py

```diff
@@ -159,6 +238,23 @@ def _postprocess_decoded_text(self, decoded_corpus: tuple[str, ...]) -> Optional
 
         In case of corrupt input arguments, None is returned
         """
+        if not isinstance(decoded_corpus, tuple) or len(decoded_corpus) == 0:
+            return None
+
+        resulting_text = ""
+        for index, token in enumerate(decoded_corpus):
+            if index == 0:
+                resulting_text += token.upper()
+            elif token == self._end_of_word_token:
+                if index == len(decoded_corpus) - 1:
+                    resulting_text = f"{resulting_text}."
+                else:
+                    resulting_text = f"{resulting_text} "
```

Comment:
save the `end_of_the_text` into a variable and only then, after if/else format the string with `end_of_the_text`

Reply:
fixed